src/zfs-9p/zfs_9p_operations.c

```diff
@@ -970,6 +986,85 @@ p9_zfs_walk(Ixp9Req *r){
 	ixp_respond(r, NULL);
 }
 
+static void
+p9_zfs_freefid(IxpFid* fid){
+	printf("p9_zfs_freefid\n");
+	printf("fid ID <%u>\n", fid->qid.path);
+	zfsAux *z = fid->aux;
+	fid->aux = NULL;
+	free(z);
+	printf("Fin freefid\n");
+}
+
+static void
+p9_zfs_remove(Ixp9Req *r){
+	printf("p9_zfs_remove ID <%u>\n", r->fid->qid.path);
+	int error;
+	zfsAux *z = r->fid->aux;
+	vnode_t *vp = z->vp;
+	
+	zfsvfs_t *zfsvfs = vfs_9p->vfs_data;
+	ZFS_ENTER_9P(zfsvfs, error);
+	if(error){
+		ixp_respond(r, "ZFS_ENTER_9PError");
+		return;
+	}
+	
+	znode_t *znode;
+
+	error = zfs_zget(zfsvfs, r->fid->qid.path, &znode, B_FALSE);
+	if(error) {
+		ZFS_EXIT(zfsvfs);
+		/* If the inode we are trying to get was recently deleted
+		   dnode_hold_impl will return EEXIST instead of ENOENT */
+		char aux[6];
+		sprintf(aux, "%d", error);
+		ixp_respond(r, error == EEXIST ? "ENOENT" : aux);
+		return;
+	}
+	
+	if(znode==NULL){
+		ixp_respond(r,"znodenil");
```

Comment:
If there is a way to make your messages match the Plan 9 strings, that's better. I've had to debug some really weird problems that were because of error messages like this.